### PR TITLE
Code refactoring: extract function

### PR DESCRIPTION
**Pull request by cyrilpangilinan**
Extracted hook setup function for code reusability

### CODE REVIEW

**Comment by zckrs** [comment]
:+1:

File: addons/expviewer/expviewer.lua [hunk 1]
```diff
@@ -514,32 +514,9 @@ function SAVE_POSITION_TO_FILE(xPosition, yPosition)
...
-
-if _G["HEADSUPDISPLAY_ON_MSG_OLD"] == nil then
-	_G["HEADSUPDISPLAY_ON_MSG_OLD"] = _G[hudHook];
-	_G[hudHook] = HEADSUPDISPLAY_ON_MSG_HOOKED;
-else
-	_G[hudHook] = HEADSUPDISPLAY_ON_MSG_HOOKED;
-end
+SETUP_HOOK(CHARBASEINFO_ON_MSG_HOOKED, "CHARBASEINFO_ON_MSG");
+SETUP_HOOK(ON_JOB_EXP_UPDATE_HOOKED, "ON_JOB_EXP_UPDATE");
+SETUP_HOOK(HEADSUPDISPLAY_ON_MSG_HOOKED, "HEADSUPDISPLAY_ON_MSG_OLD");
```
Comment (Miei) [inline addons/expviewer/expviewer.lua]: Should read
`SETUP_HOOK(HEADSUPDISPLAY_ON_MSG_HOOKED, "HEADSUPDISPLAY_ON_MSG");`

Comment (cyrilpangilinan) [inline addons/expviewer/expviewer.lua]: Oh good catch, was too busy re-organizing my local branch

**Comment by Excrulon** [comment]
This makes it a lot cleaner for now, but the problem is we can't use this method long term. As soon as someone hooks the same thing twice, that functionality will break. I've already ran into problems needing to hook the same thing multiple times in my own addons. It'll just get worse as people make more.

I have some utility code that may fix it as long as everyone uses it, I just need to test it.

**Comment by cyrilpangilinan** [comment]
Yeah I agree, just cleaned it up for now until we set a standard for hooking.

**Pull request merged**